src/network/mcpe/handler/InGamePacketHandler.php

```diff
@@ -685,6 +686,13 @@ public function handleMapInfoRequest(MapInfoRequestPacket $packet) : bool{
 	}
 
 	public function handleRequestChunkRadius(RequestChunkRadiusPacket $packet) : bool{
+		if($this->player->getViewDistance() !== $packet->radius){
+			$ev = new PlayerViewDistanceChangeEvent($this->player, $packet->radius);
```

Comment:
This is the wrong place. It won't be called if it's changed for reasons other than packet.

Reply:
So in the Player class directly? This will group together all the way to change its view distance